tlog-cosignature.md

```diff
@@ -0,0 +1,100 @@
+# Transparency Log Cosignatures
+
+A cosignature is a statement by a transparency log [witness][] that it verified
+the consistency of a [checkpoint][]. Log clients can verify a quorum of
+cosignatures to prevent split-view attacks before trusting an inclusion proof.
+
+```
+example.com/behind-the-sofa
+20852163
+CsUYapGGPo4dkMgIAUqom/Xajj7h2fB2MPA3j2jxq2I=
+
+— example.com/behind-the-sofa Az3grlgtzPICa5OS8npVmf1Myq/5IZniMp+ZJurmRDeOoRDe4URYN7u5/Zhcyv2q1gGzGku9nTo+zyWE+xeMcTOAYQ8=
+— witness.example.com/w1 jWbPPwAAAABkGFDLEZMHwSRaJNiIDoe9DYn/zXcrtPHeolMI5OWXEhZCB9dlrDJsX3b2oyin1nPZ\nqhf5nNo0xUe+mbIUBkBIfZ+qnA==
+```
+
+## Conventions used in this document
+
+Data structures are defined according to the conventions laid out in Section 3
+of [RFC 8446][].
+
+`U+` followed by four hexadecimal characters denotes a Unicode codepoint, to be
+encoded in UTF-8. `0x` followed by two hexadecimal characters denotes a byte
+value in the 0-255 range. `||` denotes concatenation.
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
+"SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this
+document are to be interpreted as described in [BCP 14][] [RFC 2119][] [RFC
+8174][] when, and only when, they appear in all capitals, as shown here.
+
+[RFC 8446]: https://www.rfc-editor.org/rfc/rfc8446.html
+[BCP 14]: https://www.rfc-editor.org/info/bcp14
+[RFC 2119]: https://www.rfc-editor.org/rfc/rfc2119.html
+[RFC 8174]: https://www.rfc-editor.org/rfc/rfc8174.html
+
+## Format
+
+Concretely, a cosignature is a [note signature][] applied to a [checkpoint][].
+
+Per the signed note format, a note signature line is
+
+    — <key name> base64(32-bit key ID || signature)
+
+The key name SHOULD be a schema-less URL that identifies the witness. Like the
+checkpoint origin line, this is for disambiguation, and MAY NOT match a publicly
+reachable endpoint (if any).
```

Comment:
Suggestion: the "(if any)" can probably be removed.

Reply:
Agreed, but since it's the only remaining editorial comment and GitHub annoyingly requires approval even when the author is an owner, punting on it to avoid another cycle of approvals.

See #58 for improving this.